napari/_vispy/layers/base.py

```diff
@@ -155,10 +155,11 @@ def _on_matrix_change(self):
 
     def _on_experimental_clipping_planes_change(self):
         if hasattr(self.node, 'clipping_planes'):
-            self.node.clipping_planes = (
-                # invert axes because vispy uses xyz but napari zyx
-                self.layer.experimental_clipping_planes.as_array()[..., ::-1]
-            )
+            prev = self.node.clipping_planes
+            # invert axes because vispy uses xyz but napari zyx
+            new = self.layer.experimental_clipping_planes.as_array()[..., ::-1]
```

Comment:
What is the cost of calculating this each time?

Reply:
hopefullly better than what we have now :P The only good solution is the fix from vispy that checks only length changes, but we can't quite do that here because we still need to be able to forward changes to the plane to vispy, even if the number of planes does not change.